src/onnx/quantize_dequantize_linear.cpp

```diff
@@ -0,0 +1,139 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <migraphx/onnx/quantize_dequantize_linear.hpp>
+#include <migraphx/ranges.hpp>
+#include <migraphx/make_op.hpp>
+#include <migraphx/tune_axis.hpp>
+#include <migraphx/common.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace onnx {
+
+void transform_quantize_dequantize_linear_inputs(const onnx_parser::node_info& info,
+                                                 const std::string& op_name,
+                                                 int block_size,
+                                                 int axis,
+                                                 std::vector<instruction_ref>& args)
+{
+    const auto x      = args.at(0);
+    const auto x_lens = x->get_shape().lens();
+    const auto x_rank = x_lens.size();
+
+    instruction_ref y_scale = args.at(1);
+    const auto y_scale_lens = y_scale->get_shape().lens();
+    const auto y_scale_rank = y_scale_lens.size();
+
+    // Per-tensor (per-layer) granularity
+    if(y_scale->get_shape().elements() == 1)
+    {
+        std::transform(args.begin() + 1, args.end(), args.begin() + 1, [&](auto ins) {
+            return info.add_instruction(make_op("multibroadcast", {{"out_lens", x_lens}}), ins);
+        });
+    }
+    // Per-axis granularity
+    else if(y_scale_rank == 1)
+    {
+        axis = tune_axis(x_rank, axis, op_name);
+        if(x_lens[axis] != y_scale_lens[0])
+        {
+            MIGRAPHX_THROW(op_name + ": For per axis granularity the length of y_scale (actual: " +
+                           to_string(y_scale_lens[0]) + ") must be equal to size of x on axis " +
+                           to_string(axis) + "(actual :" + to_string(x_lens[axis]) + ")");
+        }
+
+        std::transform(args.begin() + 1, args.end(), args.begin() + 1, [&](auto ins) {
+            return info.add_instruction(
+                make_op("broadcast", {{"axis", axis}, {"out_lens", x_lens}}), ins);
+        });
+    }
+    // Blocked granularity
+    else
+    {
+        axis = tune_axis(x_rank, axis, op_name);
+        if(block_size == 0)
```

Comment:
Our quark generated graph doesn't use an explicit `block_size`. So this assumption about its being `0` needs to be tweaked a little bit. I think this parameter is optional. So we should work well in case it isn't supplied -- and not assume it is `0` then, and its final value should be computed to be = `block_size_min`. OTOH, if `block_size` is supplied with a model, and it isn't a `0`, then we should compare it within the lower and upper bounds. Thanks.

Reply:
Therefore, please remove this exception clause.

Reply:
The ONNX spec states it is an optional attribute, with a default value of 0: 
https://onnx.ai/onnx/operators/onnx__QuantizeLinear.html#attributes

Reply:
We don't have the quark-generated graph compiling with your current code.  I can change this code later. Thanks.